studio/data/projects/project-readonly-status-query.ts

```diff
@@ -0,0 +1,55 @@
+import { useQuery, useQueryClient, UseQueryOptions } from '@tanstack/react-query'
+import { get } from 'lib/common/fetch'
+import { API_ADMIN_URL } from 'lib/constants'
+import { useCallback } from 'react'
+import { projectKeys } from './keys'
+
+export type ProjectReadOnlyStatusVariables = { projectRef?: string }
+export type ProjectReadOnlyStatus = {
+  enabled: boolean
+  override_enabled: boolean
+  override_active_until: string
+}
+
+export async function getProjectReadOnlyStatus(
+  { projectRef }: ProjectReadOnlyStatusVariables,
+  signal?: AbortSignal
+) {
+  if (!projectRef) throw new Error('Project ref is required')
+  const data = await get(`${API_ADMIN_URL}/projects/${projectRef}/readonly`, { signal })
```

Comment:
shouldn't this just come down on another query?

like if i was using this API, i'd expect to see readOnly in the response for something like /projects not have to do an individual query.

Reply:
@soedirgo mentioned in notion

"this is per-project, and each request is 2 round trips to the DB, so I don’t think we should add this to the projects endpoint by default - it’d be too slow. if FE can get the status with SWR that should be good enough?"

Reply:
<img width="1797" alt="Screenshot 2023-06-22 at 2 39 43 pm" src="https://github.com/supabase/supabase/assets/10985857/70c9454f-fe29-487b-908e-6942212fd2fa">

...did someone say GraphQL?

Reply:
If we're making the requests here anyway, they should be on the projects endpoint, even if it's a little slower :/ (I doubt it'll be harder on our API than this ⬆️)

Reply:
@soedirgo 

I agree with @alaister 

this is just kicking the same problem down to FE. and we're making the API DX worse since anyone who wants to use the rest API for checking readOnly now has to map through all their project refs and do a request for each one.

Reply:
also seems like we could just improve our DB projects view to contain readOnly ? 

I'm also thinking a really good outcome in future might be we are using realtime to listen to changes in the projects view (can you listen to a view...?) and FE can show/hide UI based on the readOnly attribute changing.

But I'm assuming here that knowing that a project instance has become 'readOnly' is an event we can accurately capture and listen to - which i'm guessing it isn't.